File: Lab4/Ruby/insertionsort.rb
def insertion_sort(arr)
    for i in 1...(arr.length)
        j = i
        while j > 0
            if arr[j-1] > arr[j]
                temp = arr[j]
                arr[j] = arr[j-1]
                arr[j-1] = temp
            else
                break
            end
            j = j - 1
        end
    end
    return arr
end

arr = [ 3, 777, 30, 40, 1, 10, 467, 333, 111]
p insertion_sort(arr)

File: Lab4/Rust/insertionsort.rs
fn main() {
    let mut v = vec![1, 3 , 7 , 40, 120, 64, 5];
    insertion_sort(&mut v);
    println!("Final sorted list: {:?}", v);
}

fn insertion_sort(arr: &mut Vec<isize>){
    for i in 1..arr.len(){
        let mut j = i;
        while j > 0 && arr[j] < arr[j-1]{
            arr.swap(j,j-1);
            j=j-1;
        }
    }
}
